### PR TITLE
Null Safety Support 1.0

### PR DESCRIPTION
**Pull request by Chever-John**
1. remove some annotations.
2. null-safety supported.
3. no '!' in static functions.

### CODE REVIEW

File: lib/record.dart [hunk 1]
```diff
@@ -251,14 +252,14 @@ class NDEFRecord {
 
   /// Decode a [NDEFRecord] record from raw data.
   static NDEFRecord doDecode(
-      TypeNameFormat tnf, Uint8List type, Uint8List payload,
-      {Uint8List id, TypeFactory typeFactory = NDEFRecord.defaultTypeFactory}) {
-    NDEFRecord record = typeFactory(tnf, utf8.decode(type));
-    if (payload.length < record.minPayloadLength) {
+      TypeNameFormat? tnf, Uint8List? type, Uint8List? payload,
```
Comment (Harry-Chen) [inline lib/record.dart:255]: Seems parameters such as `tnf`, `type` and `payload` need not to be nullable.

File: lib/record/bluetooth.dart [hunk 2]
```diff
@@ -767,19 +769,20 @@ class BluetoothEasyPairingRecord extends BluetoothRecord {
   @override
   String toString() {
     var str = "BluetoothEasyPairingRecord: ";
-    str += "address=${address.address} ";
+    // str += "address=${address?.address} ";
+    str += "address=${(address?.address) == null ? " address is null " : address?.address} ";
```
Comment (Harry-Chen) [inline lib/record/bluetooth.dart:773]: Replace with cleaner code:

```suggestion
    str += "address=${address?.addresss ?: '(null)'} ";
```

File: lib/record/bluetooth.dart [hunk 3]
```diff
@@ -911,20 +913,20 @@ class BluetoothLowEnergyRecord extends BluetoothRecord {
   String get decodedType {
     return BluetoothLowEnergyRecord.classType;
   }
-
+  /// TODO: delete '!'
```
Comment (Harry-Chen) [inline lib/record/bluetooth.dart:916]: Note this

File: lib/record/mime.dart [hunk 4]
```diff
@@ -9,18 +9,19 @@ class MimeRecord extends NDEFRecord {
...
   }
-
+  /// TODO: delete '!'
   @override
   String toString() {
     var str = "MimeRecord: ";
     str += basicInfoString;
-    str += "payload=${payload.toHexString()}";
+    // str += "payload=${payload!.toHexString()}";
+    str += "payload=${(payload?.toHexString()) ==  null ? " payload is null " : payload?.toHexString()}";
```
Comment (Harry-Chen) [inline lib/record/mime.dart:24]: See above, replace with '?:' operator

File: lib/record/wellknown.dart [hunk 5]
```diff
@@ -9,19 +9,20 @@ class WellKnownRecord extends NDEFRecord {
...
-
+  /// TODO: delete '!'
   @override
   String toString() {
     var str = "WellKnownRecord: ";
     str += basicInfoString;
     str += "type=$decodedType ";
-    str += "payload=${payload.toHexString()}";
+    // str += "payload=${payload!.toHexString()}";
+    str += "payload=${(payload?.toHexString()) ==  null ? " payload is null " : payload?.toHexString()}";
```
Comment (Harry-Chen) [inline lib/record/wellknown.dart:25]: Replace with `?:`

**Review by Harry-Chen — CHANGES_REQUESTED** [review]
Almost LGTM. Please modify according to my comments to make the code neater & cleaner.

File: lib/record/wellknown.dart (hunk 5 above)
Comment (Chever-John) [reply lib/record/wellknown.dart]: have changed to ??

**Review by Harry-Chen — APPROVED** [review]
LGTM now. Need some tiny fixes.

**Pull request merged**